src/group/mls_group/test_mls_group.rs

```diff
@@ -37,3 +43,267 @@ fn test_mls_group_persistence() {
+            CredentialBundle::new("Bob".into(), CredentialType::Basic, ciphersuite.name()).unwrap();
+
+        // Mandatory extensions
+        let capabilities_extension = Box::new(CapabilitiesExtension::new(
+            None,
+            Some(&[ciphersuite.name()]),
+            None,
+        ));
+        let lifetime_extension = Box::new(LifetimeExtension::new(60));
+        let mandatory_extensions: Vec<Box<dyn Extension>> =
+            vec![capabilities_extension, lifetime_extension];
+
+        // Generate KeyPackages
+        let alice_key_package_bundle = KeyPackageBundle::new(
+            &[ciphersuite.name()],
+            &alice_credential_bundle,
+            mandatory_extensions.clone(),
+        )
+        .unwrap();
+
+        let bob_key_package_bundle = KeyPackageBundle::new(
+            &[ciphersuite.name()],
+            &bob_credential_bundle,
+            mandatory_extensions.clone(),
+        )
+        .unwrap();
+        let bob_key_package = bob_key_package_bundle.key_package();
+
+        // === Alice creates a group ===
+        let group_id = [1, 2, 3, 4];
+        let mut group_alice = MlsGroup::new(
+            &group_id,
+            ciphersuite.name(),
+            alice_key_package_bundle,
+            GroupConfig::default(),
+        )
+        .unwrap();
+
+        // === Alice adds Bob ===
+        let bob_add_proposal = group_alice.create_add_proposal(
+            group_aad,
+            &alice_credential_bundle,
+            bob_key_package.clone(),
+        );
+        let epoch_proposals = &[&bob_add_proposal];
+        let (mls_plaintext_commit, welcome_bundle_alice_bob_option, kpb_option) = group_alice
+            .create_commit(
+                group_aad,
+                &alice_credential_bundle,
+                epoch_proposals,
+                &[],
+                false,
+            )
+            .expect("Error creating commit");
+        let commit = match mls_plaintext_commit.content() {
+            MLSPlaintextContentType::Commit((commit, _)) => commit,
+            _ => panic!("Wrong content type"),
+        };
+        assert!(!commit.has_path() && kpb_option.is_none());
+        // Check that the function returned a Welcome message
+        assert!(welcome_bundle_alice_bob_option.is_some());
+
+        group_alice
+            .apply_commit(&mls_plaintext_commit, epoch_proposals, &[])
+            .expect("error applying commit");
+        let ratchet_tree = group_alice.tree().public_key_tree_copy();
+
+        let group_bob = MlsGroup::new_from_welcome(
+            welcome_bundle_alice_bob_option.unwrap(),
+            Some(ratchet_tree),
+            bob_key_package_bundle,
+        )
+        .unwrap();
+
+        // Make sure that both groups have the same public tree
+        if group_alice.tree().public_key_tree() != group_bob.tree().public_key_tree() {
+            _print_tree(&group_alice.tree(), "Alice added Bob");
+            panic!("Different public trees");
```

Comment:
`expect()`

Reply:
Again, this copied from another test and I might as well remove it.